compiler/parser/RdrHsSyn.hs

```diff
@@ -2619,15 +2619,14 @@ mkPrimInterfaceMethod :: String -> String -> LHsBind GhcPs
 mkPrimInterfaceMethod methodName primArg =
   mkTemplateClassMethod methodName [] (mkPrimitive "primitiveInterface" primArg) Nothing
 
--- | Construct a @data X a b c = X {...} deriving (Eq, Show)@
-mkTemplateDataDecl ::
+mkDamlDataDecl ::
      SrcSpan                 -- ^ the span to associate with
-  -> Located RdrName         -- ^ template 'T' (or choice 'S')
+  -> Either (Located RdrName) (Located RdrName) -- ^ template 'T' (or choice 'S')
```

Comment:
maybe consider defining a custom ADT here, something along the lines of
```
data DataDeclName = TemplateName (Located RdrName) | ChoiceName (Located RdrName)
```
Probably easier to understand than `Either`.

Reply:
True, I'll change.